interfaces: Adds IPv6 support and 0.0.0.0 gateway

justinschw opened the pull request:
Creates an ipv6 field in interfaces if inet6 is present, and fixes crash if only ipv6 and not ipv4 is present.
Also updates the gateway parsing as on some systems the destination shows as 0.0.0.0 instead of default or link-local.

justinschw committed:
Adds IPv6 support and 0.0.0.0 gateway

justinschw commented:
I realize this change doesn't include ipv6 routes, or actually configuring ipv6 interfaces. I could add that on to this PR if you want, or put them in a separate PR.

FGRibreau commented:
> I realize this change doesn't include ipv6 routes, or actually configuring ipv6 interfaces. I could add that on to this PR if you want, or put them in a separate PR.

Yes, could you please add these test examples and I will merge it ! :+1:
Thanks a lot!

FGRibreau reviewed (changes requested):
https://github.com/FGRibreau/network-config/pull/39#issuecomment-896044169

justinschw committed:
Adds ip6 routing and configuration

justinschw commented:
Ok, the latest commit should add these features.

justinschw committed:
Makes gateway optional, adds optional network and manual config

justinschw commented:
I hope you don't mind, there are a few more useful additions; didn't mean to pile onto this PR, but I was hoping it would be OK. I wanted to make gateway optional for static IPs, and it would also be nice to have an optional network field, so I added that too. I also added a manual configuration mode.

FGRibreau commented on src/ifconfig.interfaces.js:
```diff
@@ -18,7 +19,7 @@ module.exports = function (cp) {
       }
 
-      const routeCommand = options.gateway !== undefined && options.gateway.resolveHostNames ? 'route' : 'route -n';
+      const routeCommand = options.gateway !== undefined && options.gateway.resolveHostNames ? 'route' : 'route -n && route -6n';
```
Could we allow this command `route -6n` through an option? I'm pretty sure not everyone will have the latest route version and thus have the -6n option

justinschw replied:
Added options.gateway.6n

FGRibreau commented on src/ifconfig.interfaces.js:
```diff
@@ -59,14 +60,21 @@ function parse(ifConfigOut, routeOut, interfacesContent) {
       mac: getInterfaceMacAddr(inface),
       gateway: getGateway(routeOut),
+      ...ipv6 && { ip6: ipv6 },
+      ...ip6prefixlen && {prefixlen: ip6prefixlen},
```
it should be `{ip6prefixlen: ip6prefixlen}` right?

justinschw committed:
Making requested changes per PR

justinschw committed:
Merge branch 'FGRibreau:master' into ipv6

FGRibreau commented:
Released in v3.0.0 thanks @justinschw !
